src/nidigital/unit_tests/test_nidigital.py

```diff
@@ -250,3 +260,61 @@ def test_fetch_history_ram_cycle_information_samples_to_read_all(self):
+        self.patched_library.niDigital_FetchHistoryRAMCycleInformation.side_effect = self.side_effects_helper.niDigital_FetchHistoryRAMCycleInformation
+        self.side_effects_helper['FetchHistoryRAMCycleInformation']['patternIndex'] = 0
+        self.side_effects_helper['FetchHistoryRAMCycleInformation']['timeSetIndex'] = 0
+        self.side_effects_helper['FetchHistoryRAMCycleInformation']['vectorNumber'] = 0
+        self.side_effects_helper['FetchHistoryRAMCycleInformation']['cycleNumber'] = 0
+        self.side_effects_helper['FetchHistoryRAMCycleInformation']['numDutCycles'] = 1
+        self.patched_library.niDigital_GetPatternName.side_effect = self.side_effects_helper.niDigital_GetPatternName
+        self.side_effects_helper['GetPatternName']['name'] = 'new_pattern'
+        self.patched_library.niDigital_GetTimeSetName.side_effect = self.side_effects_helper.niDigital_GetTimeSetName
+        self.side_effects_helper['GetTimeSetName']['name'] = 't0'
+        self.patched_library.niDigital_FetchHistoryRAMScanCycleNumber.side_effect = self.side_effects_helper.niDigital_FetchHistoryRAMScanCycleNumber
+        self.side_effects_helper['FetchHistoryRAMScanCycleNumber']['scanCycleNumber'] = -1
+        self.patched_library.niDigital_FetchHistoryRAMCyclePinData.side_effect = self.niDigital_FetchHistoryRAMCyclePinData_check_pins_looping
+        with nidigital.Session('') as session:
+            self.expected_pin_list_check_pins_looping = 'PinA,PinB'
+            session.sites[0].pins['PinA', 'PinB'].fetch_history_ram_cycle_information(position=0, samples_to_read=-1)
+            self.expected_pin_list_check_pins_looping = ''
+            session.sites[0].fetch_history_ram_cycle_information(position=0, samples_to_read=-1)
+            assert self.patched_library.niDigital_FetchHistoryRAMCyclePinData.call_count == 4
+
+    # Helper function for validating site behavior in fetch_hram.
+    def niDigital_GetHistoryRAMSampleCount_check_site_looping(self, vi, site, sample_count):  # noqa: N802
+        assert site.value.decode('ascii') == 'site{}'.format(self.site_vals_looping[self.iteration_check_site_looping])
+        sample_count.contents.value = 0  # we don't care if this is right as long as the fetch does not error
+        self.iteration_check_site_looping += 1
+        return 0
+
+    def test_fetch_history_ram_cycle_information_site_n(self):
+
+        self.patched_library.niDigital_GetHistoryRAMSampleCount.side_effect = self.niDigital_GetHistoryRAMSampleCount_check_site_looping
+        self.side_effects_helper['GetHistoryRAMSampleCount']['sampleCount'] = 1
+
+        with nidigital.Session('') as session:
+            for s in self.site_vals_looping:
+                session.sites[s].fetch_history_ram_cycle_information(position=0, samples_to_read=0)
+            self.patched_library.niDigital_GetHistoryRAMSampleCount.assert_called()
```

Comment:
Why not `call_count == len(self.site_vals_looping)`?

Reply:
Because the value that controlled the number was elsewhere and the count wasn't that important.  I can change it if you want, though.

Reply:
You have:
```
            assert self.patched_library.niDigital_FetchHistoryRAMCyclePinData.call_count == 4
```
in the previous test.

I don't mind either way, as long as the two tests are consistent.

Reply:
Done.